packages/asset-swapper/src/utils/market_operation_utils/pools_cache/balancer_v2_utils.ts

```diff
@@ -29,8 +70,74 @@ export class BalancerV2PoolsCache extends PoolsCache {
     //     }
     // }
 
+    protected async _fetchTopPoolsAsync(): Promise<BalancerPoolResponse[]> {
+        const query = gql`
+            query fetchTopPools($topPoolsFetched: Int!) {
+                pools(
+                    first: $topPoolsFetched
+                    where: { totalLiquidity_gt: 0 }
+                    orderBy: swapsCount
+                    orderDirection: desc
+                ) {
+                    id
+                    swapFee
+                    totalWeight
+                    tokensList
+                    amp
+                    totalShares
+                    tokens {
+                        id
+                        address
+                        balance
+                        decimals
+                        symbol
+                        weight
+                    }
+                }
+            }
+        `;
+
+        const { pools } = await request<{ pools: BalancerPoolResponse[] }>(this.subgraphUrl, query, {
+            topPoolsFetched: this._topPoolsFetched,
+        });
+
+        return pools;
+    }
+    protected async _loadTopPoolsAsync(): Promise<void> {
+        const fromToPools: {
+            [from: string]: { [to: string]: Pool[] };
+        } = {};
+
+        const pools = await this._fetchTopPoolsAsync();
+        pools.forEach(pool => {
+            const { tokensList } = pool;
+            for (const from of tokensList) {
+                for (const to of tokensList.filter(t => t.toLowerCase() !== from.toLowerCase())) {
+                    if (!fromToPools[from]) {
+                        fromToPools[from] = {};
+                    }
+                    if (!fromToPools[from][to]) {
+                        fromToPools[from][to] = [];
+                    }
+                    try {
+                        // The list of pools must be relevant to `from` and `to`  for `parsePoolData`
+                        const [poolData] = parsePoolData({ [pool.id]: pool as any }, from, to);
+                        fromToPools[from][to].push(
+                            BalancerV2PoolsCache._parseSubgraphPoolData(poolData[pool.id], from, to),
+                        );
+                        // Cache this as we progress through
+                        const expiresAt = Date.now() + this._cacheTimeMs;
+                        this._cachePoolsForPair(from, to, fromToPools[from][to], expiresAt);
+                    } catch (err) {
+                        this._warningLogger(err, `Failed to load Balancer V2 top pools`);
+                        // soldier on
+                    }
+                }
+            }
+        });
+    }
     protected async _fetchPoolsForPairAsync(takerToken: string, makerToken: string): Promise<Pool[]> {
-        const query = `
+        const query = gql`
```

Comment:
oh, never seen this syntax before, how does this work?

Reply:
I think it's just a tagged string template so it doesn't do so much. However, now that the editor understands that it's a GQL query you can get prettier formatting and some basic linting :)